wordpress-mobile/WordPress-Android
# Issue/mark blogging prompt as answered

This PR adds ability to mark prompt as answered.

Corresponding Flux-C PR is [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2422).

To test DB migration:
- Install the app version from trunk branch and login into it.
- Install version from this branch over version from trunk and confirm that app works and DB migration is successful.
- Navigate to Post screen and try publishing and trashing posts. Make sure app does not crash.

To test:
- Using wasabi version of the app enable BP feature flag and restart the app.
- Using prompt card from My Site fragment answer it and publish post.
- If you are on My Site screen, confirm that the BP card is updated after post finished uploading.
- Confirm that the compact card shows prompt as Answered as well.


Note:
- You can delete post with BP answer, and it should mark prompt as unanswered.

## Regression Notes
1. Potential unintended areas of impact
- Failed DB migration.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
- Various automated tests.


PR submission checklist:
